### ISSUE TITLE
Why the blkio_delay is so different with iotop-py?

### ISSUE THREAD

**Issue by zylthinking** [Tomas-M/iotop#65]
I made a busy reading of disk in clickhouse, and notice the GRAPH[IO]▽   column in the view is quite different with the IO> column in iotop.

The main thread of clickhouse shows the  GRAPH[IO] is 100% and  IO> is zero.
Which I think iotop is correct.

Because the IO mainly happens in a background thread, the main thread should be idle.

I see the data is fetched from netlink instead of /proc;  but the algorithm is similar. 
I don't understand the result, is there something I missed that GRAPH[IO] is not equivalent to IO> ?

**Comment by zylthinking** [comment]
I found the logic, you added all threads statistics to main threads besides to thread itself.

**Comment by zylthinking** [comment]
Yes, it should maintain the process level information independently.

And I found another possible problem, `pidgen_cb` may enumerate other threads than main thread, that makes losing of some thread's data at process level.

**Comment by bbonev** [comment]
I will separate the information. But if there is `average`/`max` then the memory usage will definitely increase. My thinking is that memory nowadays is cheap and saving some cycles should be preferred. Calculating the possible aggregates should definitely happen during the data collection and not during the display (unless the possible views count increases way too much). And if there is more than one 'view', the collection should maintain the possible data sets to enable toggling between them without starting over with the history data when the view is changed.

After checking the code, I can confirm that it is possible `pidgen_cb` to see a thread before the main thread and then it will not sum its data into the main thread. In thread mode, the display will be correct besides that the main thread has summed data instead of its own only.

The aggregation function currently in use should have been `average`. Since IO is measured as time, summing the threads can go over 100% and that would later be clipped to 100%...

Maybe adding a toggle to switch between `max` and `average` is a good idea. Of course that applies to process mode only.

To summarize:

**Bug 1)** Do not aggregate the data in the main thread [thread mode only]; keep both aggregated main thread data and its original data to enable proper switching between thread and process mode.
**Bug 2)** Change `pidgen_cb` to properly handle any order of the threads.
**Bug 3)** Actually do the aggregation instead of summing the data.
**Feat 1)** Add a toggle to use `max`/`average` as aggregate function.

**Comment by bbonev** [comment]
Besides aggregation of IO and SWAPIN are in the latest git. Please test how it goes with this change.

The aggregation of IO and SWAPIN is coming in couple of days

**Comment by bbonev** [comment]
Somehow I do not see how aggregating IO and SWAPIN with an `AVERAGE` function makes any sense for the viewer.

In case the threaded application has many sleeping threads, the result will become way too low. This will hide the result from the single thread waiting 100% for IO...

That is why I am making it fixed to `MAX`.
